tests/test_pages_price.py

```diff
@@ -0,0 +1,244 @@
+import pytest
+from web_poet import HttpResponse, field
+
+from zyte_common_items import ProductPage
+
+
+@pytest.mark.asyncio
+async def test_price_selector():
+    class CustomProductPage(ProductPage):
+        call_count = 0
+
+        @field
+        def price(self):
+            self.call_count += 1
+            return self.css("div::text")
+
+    html = b"""
+    <!DOCTYPE html>
+    <html>
+        <body>
+            <div>$13.2</div>
+        </body>
+    </html>
+    """
+    url = "https://example.com"
+    page = CustomProductPage(response=HttpResponse(url=url, body=html))
+    assert page.call_count == 0
+    assert page.price == "13.2"
+    assert page.call_count == 1
+    assert page.currency is None
+    assert await page.currencyRaw == "$"
+    assert page.call_count == 1
+
+    # access currency fields before the price field
+    page = CustomProductPage(response=HttpResponse(url=url, body=html))
+    assert page.call_count == 0
+    assert page.currency is None
+    assert await page.currencyRaw == "$"
+    assert page.call_count == 1
+    assert page.price == "13.2"
+    assert page.call_count == 2
```

Comment:
Even if the reason for this behavior is clear, I think it could make sense to mark this particular "2" as xfail, and open an issue in the repo. I'm fine with merging without addressing it now.

End-to-end the behavior doesn't make much sense; we might be calling price extraction multiple times when to_item is called.

Reply:
After adding call count checks to `test_price_explicit` we now have 3 asserts in total that should be xfailed, but I'm not sure how to cleanly extract them into tests that can be marked with xfail without duplicating all of them.